pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix

```diff
@@ -51,6 +69,11 @@ stdenv.mkDerivation rec {
     done
   '';
 
+  preFixup = ''
+    # fix missing libudev
+    ln -s ${lib.getLib systemd}/lib/libudev.so.1 $out/lib/libudev.so.1
+  '';
```

Comment:
This should probably be fixed with patchelf. Also this likely creates conflicts when combined in one environment.

Reply:
what is PatchIf? 

Thank you for all the help! much appreciated

Reply:
patchelf is a tool to manipulate .so files. You probably want something like https://github.com/NixOS/nixpkgs/blob/c48a958765d67d6f7e74ad8ab194440c3e8d1d08/pkgs/applications/editors/emacs/generic.nix#L239